aten/src/ATen/native/transformers/cuda/sdp_utils.h

```diff
@@ -73,7 +73,7 @@ inline bool check_for_attn_mask(sdp_params params, bool debug) {
 inline bool check_tensor_shapes(sdp_params params, bool debug) {
   auto query_dim = params.query.dim();
   if (!(query_dim == params.key.dim() && query_dim == params.value.dim() &&
-        query_dim == 4)) {
+        (query_dim == 4 || query_dim == 3))) {
```

Comment:
This should only be 4 but because of where we do the chcek for transformer encoder_layer we get dim=3 so I just added this for now

Reply:
view and then check instead of making the check brittle